### ISSUE TITLE
Scroll Issue

### ISSUE THREAD

**Issue by coldily** [seiyria/bootstrap-slider#755]
Issues
=============
**NOTE:** When "$(document).scrollLeft()" is not zero, then can not make sure it is correct, please fix line number 1704 code from "var distanceToSlide = eventPosition - sliderOffset;" to "var distanceToSlide = eventPosition - sliderOffset - $(document).scrollLeft();" in "bootstrap-slider.js" file or other valid solution.

Thanks.

**Comment by T-vK** [comment]
I totally agree this needs to be fixed. It has been driving me crazy for quite some time now. However, you need to take into consideration if the slider is vertical or horizontal. And then use `scrollLeft()` or `scrollTop()` accordingly. There might be [other potentially better ways](https://stackoverflow.com/questions/3234256/find-mouse-position-relative-to-element) of fixing this.

**Comment by seiyria** [comment]
We would happily accept a pull request to fix this.

**Comment by T-vK** [comment]
@seiyria 
What exactly are the tests from `ScrollableBodySpec.js` testing? After applying the fix, these tests are failing: 

>  Scrollable body test
>    Vertical scrolled body
>      X slides up when handle moves upwards after scroll page down
>        Expected 0 to equal 4. (1)
>      X slides down when handle moves downwards after scroll page down
>        Expected 0 to equal 14. (1)

**Comment by coldily** [comment]
@T-vK 
Thanks for your suggestion, I found this issue when using a horizontal slider in my project, and test it in vertical status. But only the horizontal slider has this issue, I don't thoroughly understand the slider code, maybe "sliderOffset" field calculated “$(document).scrollTop()” value, I just fix it can work properly in my project, if you have other potentially better ways, please help me fix it, thanks.

**Comment by narayanjoshi** [comment]
@coldily 
`var scrollDelta = 0, parentEl = this.sliderElem.parentNode;
while (!scrollDelta && parentEl && (parentEl.tagName.toLowerCase() !== 'html')) {
	if ((this.options.orientation === 'vertical') && (parentEl.scrollTop > 0)) {
		scrollDelta = parentEl.scrollTop;
	} else if ((this.options.orientation === 'horizontal') && (parentEl.scrollLeft > 0)) {
		scrollDelta = parentEl.scrollLeft;
	}
	if (!scrollDelta) { parentEl = parentEl.parentNode; }
}
var eventPosition = ev[this.mousePos] + scrollDelta;`

The above block fixes the issue

@seiyria 
However the test cases are failing, could you please re-visit the test cases?

Scrollable body test
   Vertical scrolled body
   Horizontal scrolled body
 Scrollable test
   Vertical inside scrollable container

**Comment by narayanjoshi** [comment]
@seiyria refer my PR which I closed because test cases were failing - 
PR https://github.com/seiyria/bootstrap-slider/pull/928

**Comment by seiyria** [comment]
If the test cases aren't working that's on you to fix.
